### PR TITLE
[BPK-1151] Add flow types decision

### PR DESCRIPTION
**Pull request by matthewdavidson**
**Note:** this cannot be merged until #397 is merged.

### CODE REVIEW

File: decisions/TEMPLATE.md [hunk 1]
```diff
@@ -2,12 +2,12 @@
 
 # Title <!-- e.g. "Squad colour coordination" -->
 
-## Decision
-<!-- e.g. "On Wednesdays we wear pink" -->
-
 ## TL;DR
 <!-- Only if long enough to merit one. -->
 
+## Decision
```
Comment (matthewdavidson) [inline decisions/TEMPLATE.md]: I moved TL;DR to top because that's where I thought it should be.

File: decisions/adding-types.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Adding types
+
+## Decision
+All components should have flow types _as well as_ `prop-types`. If you come across a file or component that isn't flow
+typed, submit a separate PR to add types to it (alongside any associated tests or `index.js` entry points). Don't
```
Comment (matteo-hertel) [inline decisions/adding-types.md]: I would add story files to the list as well

File: decisions/adding-types.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+include the type changes in the same PR as the original work.
+
+## Thinking
+Consumers might not have adopted flow yet, meaning we still want them to benefit from `prop-types` run time warnings.
+
+Regarding the idea behind separate PR's - this is simply because we want to keep them small and we feel that including
```
Comment (matteo-hertel) [inline decisions/adding-types.md]: 🎖

File: decisions/adding-types.md (hunk 2 above)
Comment (matthewdavidson) [reply decisions/adding-types.md]: good shout!

File: decisions/adding-types.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+## Decision
+All components should have flow types _as well as_ `prop-types`. If you come across a file or component that isn't flow
+typed, submit a separate PR to add types to it (alongside any associated tests, `index.js` entry points or `stories.js`
+files). Don't include the type changes in the same PR as the original work.
+
+## Thinking
+Consumers might not have adopted flow yet, meaning we still want them to benefit from `prop-types` run time warnings.
+
+Regarding the idea behind separate PR's - this is simply because we want to keep them small and we feel that including
```
Comment (shaundon) [inline decisions/adding-types.md]: `PR's` should be `PRs`

**Pull request merged**